lib/src/main/java/com/iwillfailyou/javaparser/Caller.java

```diff
@@ -0,0 +1,7 @@
+package com.iwillfailyou.javaparser;
+
+import com.github.javaparser.ast.Node;
+
+public interface Caller {
```

Comment:
Didn't have a better naming idea

Reply:
The main idea of this interface - provide a node, which would be found by some algorithm (defined in concrete implementation). May be `interface Search<T extends Node> { List<T> find(); }` and implementation `class ParentMethodOrLambda implements Search<NodeWithParameters<?>>`

Reply:
fixed. went for `Optional<T> find();` instead if `List`.